### ISSUE TITLE
Better reference formatting

### ISSUE THREAD

**Issue by dnnrly** [dnnrly/httpref#34]
**Is your feature request related to a problem? Please describe.**

Formatting of the output in this tool is rather cumbersome. It would be great if we could use a modern CLI formatting library to do something better, perhaps using something like this:
https://github.com/charmbracelet/lipgloss

**Describe the solution you'd like**
Better control over the look and feel of the output. Perhaps some additional configuration to customise it for user needs.

**Comment by georgettica** [comment]
@dnnrly could you assign me?
I will.tinker with it in the coming days and see if I can add some knobs like:
- format options (json/yaml)
- Golang template printing (similar to docker ps --format)

I will also fig deeper into lipgloss to see if it fits the bill

**Comment by georgettica** [comment]
I think I understand what you mean by reference formatting (without looking at lipgloss too much)

I think I can whip up a PR and some pictures and I'll ping you here

**Comment by georgettica** [comment]
---
To be clear, will start the effort after being assigned :/

**Comment by dnnrly** [comment]
I don't typically assign issues as I've had problems with people not completing them before. But I've had some good quality PRs from you before so happy to.

**Comment by georgettica** [comment]
❤️

**Comment by dnnrly** [comment]
> will take inspiration from https://github.com/aarol/goclean/blob/master/view.go, unless you have a better way of formatting

Yes, this is exactly the sort of thing I was thinking.

**Comment by dnnrly** [comment]
Nah, you're good. You've already done loads. Getting a designer involved is good. We need to focus on make it as easy and useful as possible.
